tofhir-server/src/main/scala/io/tofhir/server/service/mappingcontext/MappingContextFolderRepository.scala

```diff
@@ -0,0 +1,179 @@
+   * @param content   mapping context content to save
+   * @return
+   */
+  override def saveMappingContextContent(projectId: String, id: String, content: Source[ByteString, Any]): Future[Unit] = {
+    if (!mappingContextDefinitions.contains(projectId) || !mappingContextDefinitions(projectId).contains(id)) {
+      throw ResourceNotFound("Mapping context does not exists.", s"A mapping context with id $id does not exists in the mapping context repository at ${FileUtils.getPath(mappingContextRepositoryFolderPath).toAbsolutePath.toString}")
+    }
+    // Write content to the related file in the repository
+    getFileForMappingContext(projectId, id).map(file => {
+      FileOperations.saveFileContent(file, content)
+    })
+  }
+
+  /**
+   * Get the mapping context content by its id
+   *
+   * @param projectId project id the mapping context belongs to
+   * @param id        mapping context id
+   * @return
+   */
+  override def getMappingContextContent(projectId: String, id: String): Future[Source[ByteString, Any]] = {
+    if (!mappingContextDefinitions.contains(projectId) || !mappingContextDefinitions(projectId).contains(id)) {
+      throw ResourceNotFound("Mapping context does not exists.", s"A mapping context with id $id does not exists in the mapping context repository at ${FileUtils.getPath(mappingContextRepositoryFolderPath).toAbsolutePath.toString}")
+    }
+    // Read content from the related file in the repository
+    getFileForMappingContext(projectId, id).map(file => {
+      FileIO.fromPath(file.toPath)
+    })
+  }
+
+  /**
+   * Gets the file for the given mapping context id.
+   *
+   * @param fhirMappingContextId
+   * @return
+   */
+  private def getFileForMappingContext(projectId: String, fhirMappingContextId: String): Future[File] = {
+    val projectFuture: Future[Option[Project]] = projectFolderRepository.getProject(projectId)
+    projectFuture.map(project => {
+      val file: File = FileUtils.getPath(mappingContextRepositoryFolderPath, project.get.id, fhirMappingContextId).toFile
+      // If the project folder does not exist, create it
+      if (!file.getParentFile.exists()) {
+        file.getParentFile.mkdir()
+      }
+      file
+    })
+  }
+
+  /**
+   * Initializes the mapping context map from the repository
+   * Map of (project id -> mapping context list)
+   * @param mappingContextRepositoryFolderPath path to the mapping context repository
+   * @return
+   */
+  private def initMap(mappingContextRepositoryFolderPath: String): mutable.Map[String, Seq[String]] = {
+    val map = mutable.Map.empty[String, Seq[String]]
+    val folder = FileUtils.getPath(mappingContextRepositoryFolderPath).toFile
+    if (!folder.exists()) {
+      folder.mkdirs()
+    }
+    var directories = Seq.empty[File]
+    try {
+      directories = folder.listFiles.filter(_.isDirectory).toSeq
+    } catch {
+      case e: Throwable => throw FhirMappingException(s"Given folder for the mapping context repository is not valid.", e)
```

Comment:
Out of curiosity, do we really need try/catch block here, when does it throw an exception ?

Reply:
I guess you're right, we can omit that part. Also, we have same logic for other repositories (jobRepository, mappingRepository etc.), I will update those parts too.